src/server/active_tran_server.cpp

```diff
@@ -125,16 +125,41 @@ active_tran_server::init_page_server_hosts (const char *db_name)
 {
   assert_is_active_tran_server ();
 
+  /*
+   * Specified behavior:
+   * ===============================================================================
+   * |       \    hosts config     |   empty   |    bad    |          good         |
+   * |--------\--------------------|-----------|-----------|------------|----------|
+   * | storage \ connections to PS |           |           |    == 0    |   > 0    |
+   * |==========\==============================|===========|============|==========|
+   * |   local  |                      OK      |    N/A    |     OK     |   OK     |
+   * |----------|------------------------------|-----------|------------|----------|
+   * |   remote |                     Error    |   Error   |   Error    |   OK     |
+   * ===============================================================================
+   */
+
+  // read raw config
+  //
   std::string hosts = prm_get_string_value (PRM_ID_PAGE_SERVER_HOSTS);
+  m_has_remote_storage = prm_get_bool_value (PRM_ID_REMOTE_STORAGE);
 
+  // check config validity
+  //
   if (!hosts.length ())
     {
-      // no page server
-      return NO_ERROR;
+      if (m_has_remote_storage)
+	{
+	  er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_INCONSISTENT_REMOTE_STORAGE_VS_HOSTS, 0);
```

Comment:
The problem qualifies as error.
```suggestion
	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_INCONSISTENT_REMOTE_STORAGE_VS_HOSTS, 0);
```

Reply:
Done